package/MDAnalysis/topology/tpr/obj.py

```diff
@@ -31,7 +31,7 @@
 
 TpxHeader = namedtuple(
     "TpxHeader", [
-        "number", "ver_str", "precision",
```

Comment:
`number` did not correspond to anything in the TPR format.As a few other readings, it was due to a mismatch between how python reads a string in a XDR file, and how gromacs does it. Python reads the string from the current position to the first null character. Instead, gromacs reads the size as an unsigned int and read that many character. As the TPR files are written by gromacs, all strings are prefixed with there size.

Zhuyi Xue who wrote the parser, understood that an int should be read at some specific point, but did not understand why. This lead to a working parser, and some confused comments.I finally found where it comes from and moved the reading of an int, part of the reading of the string in `do_string`.

I removed the `number` field as it meant nothing (except for the length of the version string).